### PR TITLE
Clean up E0741 error explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0741.md [hunk 1]
```diff
@@ -9,12 +10,15 @@ struct A;
 struct B<const X: A>; // error!
 ```
 
-To fix this example, we derive `PartialEq` and `Eq`.
+Only structural-match types (that is, types that derive `PartialEq` and `Eq`)
+may be used as the types of const generic parameters.
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0741.md:14]:
```suggestion
may be used as the type of const generic parameters.
```

Is using `type` here more correct?

Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0741.md:14]: I \*think\* the current wording is correct. It's actually hard to phrase it haha.

**Review by pickfire — APPROVED** [review]
Looks weird to me.

**Pull request merged**